sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -1043,6 +1044,22 @@ object TypeCoercion {
       }
     }
   }
+
+  /**
+   * A special rule to support string literal as the second argument of date_add/date_sub functions,
+   * to keep backward compatibility.
+   * TODO: revisit the type coercion rules for string.
```

Comment:
How about filing a jira for this TODO then leave a jira ID here?

Reply:
AFAIK we started the ANSI type coercion (you have a design doc right?). Did we create a JIRA ticket at that time?

Reply:
Ur, I wrote the design doc though, I didn't file a jira at that time. Probably, @gengliangwang  migith have done so?